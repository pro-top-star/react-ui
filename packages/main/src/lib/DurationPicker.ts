import { DurationPicker } from '../webComponents/DurationPicker';
import type { DurationPickerPropTypes } from '../webComponents/DurationPicker';

export { DurationPicker };
export type { DurationPickerPropTypes };
