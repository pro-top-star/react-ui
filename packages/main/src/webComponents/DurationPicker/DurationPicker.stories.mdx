import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { DurationPicker } from '@ui5/webcomponents-react/lib/DurationPicker';
import { CSSProperties } from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="UI5 Web Components / DurationPicker"
  component={DurationPicker}
  argTypes={{
    ...createSelectArgTypes({}),
    valueStateMessage: { control: { disable: true } },
    ...DocsCommonProps
  }}
  args={{
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <DurationPicker {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />
<div style={{ fontFamily: 'var(--sapFontFamily)', fontSize: 'var(--sapFontSize)', color: 'var(--sapTextColor)' }}>
  <h3>Usage</h3>
  The Duration Picker is used for input of time. Users are able to select hours, minutes and seconds. The user can enter
  a time by:
  <ul>
    <li>Using the sliders that opens in a popup</li>
    <li>Typing it in directly in the input field</li>
  </ul>
  <br />
  <br />
  When the user makes an entry and chooses the enter key, the sliders shows the corresponding time. When the user directly
  triggers the sliders display, the actual time is displayed.
</div>
