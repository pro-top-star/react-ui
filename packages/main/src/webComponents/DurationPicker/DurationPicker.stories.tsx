import { action } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import { DurationPicker } from '@ui5/webcomponents-react/lib/DurationPicker';
import React from 'react';

export default {
  title: 'UI5 Web Components / DurationPicker',
  component: DurationPicker
};

export const generatedDefaultStory = () => (
  <DurationPicker
    disabled={boolean('disabled', false)}
    maxValue={text('maxValue', '23:59:59')}
    readonly={boolean('readonly', false)}
    showSeconds={boolean('showSeconds', false)}
    value={text('value', '00:00:00')}
    valueState={text('valueState', 'None')}
    onChange={action('onChange')}
  />
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
