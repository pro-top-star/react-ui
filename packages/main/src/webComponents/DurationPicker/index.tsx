import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import '@ui5/webcomponents/dist/DurationPicker';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface DurationPickerPropTypes extends Omit<WithWebComponentPropTypes, 'onChange'> {
  /**
   * Determines whether the <code>ui5-duration-picker</code> is displayed as disabled.
   */
  disabled?: boolean;
  /**
   * Defines a formatted maximal time that the user will be able to adjust.
   */
  maxValue?: string;
  /**
   * Determines whether the <code>ui5-duration-picker</code> is displayed as readonly.
   */
  readonly?: boolean;
  /**
   * Defines whether a slider for secconds will be available. By default there are sliders for hours and minutes only.
   */
  showSeconds?: boolean;
  /**
   * Defines a formatted time value.
   */
  value?: string;
  /**
   * Visualizes the validation state of the Web Component, for example <code>Error</code>, <code>Warning</code> and <code>Success</code>.<br/><br/>
   *
   * <ul> <li><code>None</code></li> <li><code>Error</code></li> <li><code>Warning</code></li> <li><code>Success</code></li> <li><code>Information</code></li> </ul>
   */
  valueState?: string;
  /**
   * Fired when the input operation has finished by pressing Enter or on focusout.
   */
  onChange?: (event: CustomEvent<{}>) => void;
}

/**
 * <code>import { DurationPicker } from '@ui5/webcomponents-react/lib/DurationPicker';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/DurationPicker" target="_blank">UI5 Web Components Playground</a>
 */
const DurationPicker: FC<DurationPickerPropTypes> = withWebComponent<DurationPickerPropTypes>(
  'ui5-duration-picker',
  ['maxValue', 'value', 'valueState'],
  ['disabled', 'readonly', 'showSeconds'],
  [],
  ['change']
);

DurationPicker.displayName = 'DurationPicker';

DurationPicker.defaultProps = {
  disabled: false,
  maxValue: '00:00:00',
  readonly: false,
  showSeconds: false,
  value: '00:00:00',
  valueState: 'None'
};

export { DurationPicker };
