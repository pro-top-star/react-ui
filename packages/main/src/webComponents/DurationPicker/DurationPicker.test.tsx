import { render } from '@shared/tests';
import { DurationPicker } from '@ui5/webcomponents-react/dist/DurationPicker';
import React from 'react';

describe('DurationPicker', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<DurationPicker />);
    expect(asFragment()).toMatchSnapshot();
  });
});
