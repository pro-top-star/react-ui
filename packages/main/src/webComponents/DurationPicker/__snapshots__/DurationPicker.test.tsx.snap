// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DurationPicker Basic Test (generated) 1`] = `
<ui5-duration-picker
  value="00:00:00"
  value-state="None"
/>
`;
